td-logger.gemspec

```diff
@@ -32,8 +32,9 @@ EOF
   gem.require_paths = ['lib']
 
   gem.add_dependency "msgpack", ">= 0.5.6", "< 2.0"
-  gem.add_dependency "td-client", ">= 0.8.66", "< 2.0"
+  gem.add_dependency "td-client", ">= 0.8.66", "< 3.0"
   gem.add_dependency "fluent-logger", ">= 0.5.0", "< 2.0"
+  gem.add_dependency "mutex_m", ">= 0.2.0", "< 1.0"
```

Comment:
`mutex_m` was independent as a Ruby gem from Ruby.
https://www.ruby-lang.org/en/news/2023/12/11/ruby-3-3-0-rc1-released/

Reply:
I'm sorry, but I don't understand why this added line is needed? If we're targeting ruby 2.7..5 and up here, IIUC, user should be able to use their own mutex_m without issue, no?

Reply:
@tung-vu-td Sorry for my late response. mutex_m will be apart from Ruby as a standard library.
If I removed the line, test case (Ruby 3.4+) got an error.
https://github.com/treasure-data/td-logger-ruby/actions/runs/9186667898/job/25262771815

Since this library uses `mutex` process, I believe this gem should be needed.
https://github.com/treasure-data/td-logger-ruby/blob/d90b9d12385eae1a4afb830e9eb12dd31bae2cde/lib/td/logger/td_logger.rb#L164